build_system/clerk_rules.ml

```diff
@@ -1070,18 +1072,25 @@ let with_ninja_process
   let ninja_process nin_file nin_fd =
     let args =
       let ninja_flags =
-        if quiet && Lazy.force ninja_version >= [1; 12] then
-          "--quiet" :: ninja_flags
+        (* Newer versions of ninja have a flag to not print "nothing to do", we
+           use that if available. *)
+        if (not Global.options.debug) && Lazy.force ninja_version >= [1; 12]
+        then "--quiet" :: ninja_flags
         else ninja_flags
       in
       ("-f" :: nin_file :: ninja_flags)
       @ if Catala_utils.Global.options.debug then ["-v"] else []
     in
     let cmdline = ninja_exec :: args in
     Message.debug "executing '%s'..." (String.concat " " cmdline);
+    let out =
+      (* In --quiet, we redirect all ninja's output to /dev/null *)
+      if quiet then Unix.descr_of_out_channel (open_out Filename.null)
```

Comment:
You never close this out channel (also, it could be better to use `Unix.openfile` directly to avoid allocating unused buffers)